indexing-hadoop/src/main/java/io/druid/indexer/InputRowSerde.java

```diff
@@ -63,9 +230,25 @@
       WritableUtils.writeVInt(out, dimList.size());
       if (dimList != null) {
         for (String dim : dimList) {
-          List<String> dimValues = row.getDimension(dim);
-          writeString(dim, out);
-          writeStringArray(dimValues, out);
+          IndexSerdeTypeHelper typeHelper = typeHelperMap.get(dim);
+          if (typeHelper == null) {
+            typeHelper = STRING_HELPER;
+          }
+          try {
+            byte[] value = typeHelper.serialize(row.getRaw(dim));
+
+            writeString(dim, out);
+            WritableUtils.writeVInt(out, typeHelper.getType().ordinal());
+            writeBytes(value, out);
+          }
+          catch (ParseException pe) {
+            if (reportParseExceptions) {
+              throw pe;
+            } else {
+              // discard the row if there was a parse error in a dimension
```

Comment:
I think it would be better to just use a "default" value for this column like zero. It's more in line with what aggregators do (see the below log message "Encountered parse error, skipping aggregator"). That way, at least we retain _something_ from the row (namely: all other columns).